### DISCUSSION TITLE
LSP keymaps missing?

### DISCUSSION THREAD

**Discussion by ruiiiijiiiiang** [LazyVim/LazyVim#4977] [Q&A] [answered]
Hi all, 

I just realized that I can't do `<leader>cr` to rename variables even though manually triggering the rename with `:lua vim.lsp.buf.rename()` works. Then I looked into the default [keymaps.lua](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/keymaps.lua) and found no mention of any LSP key bindings. Shouldn't these keymaps work out of the box as suggested on the [keymaps page](https://www.lazyvim.org/keymaps#lsp)? Do I need to manually add them to my own keymaps file?

**Comment by folke** [comment]
They are automatically loaded when an LSP attaches: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/keymaps.lua

If that doesn't work for you then you must've messed something up in your config

**Reply by ruiiiijiiiiang** [reply]
> They are automatically loaded when an LSP attaches: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/keymaps.lua
> 
> If that doesn't work for you then you must've messed something up in your config

Thank you for the response. I was assuming that since I could manually call `rename()`, that must mean my LSP was properly attached? I can also see them listed with `:LspInfo`:
![snapshot_2024-12-09_14-09-27](https://github.com/user-attachments/assets/8829c082-c5ee-4857-a9c9-30b727582e5a)

This is how I set up lspconfig:
```
	{
		"neovim/nvim-lspconfig",
		version = "*",
		dependencies = {
			"williamboman/mason.nvim",
			"williamboman/mason-lspconfig.nvim",
		},
		config = function()
			local mason = require("mason")
			local lspconfig = require("lspconfig")
			mason.setup()
			lspconfig.ts_ls.setup({})
			lspconfig.vtsls.setup({})
			lspconfig.eslint.setup({
				settings = {
					useFlatConfig = true,
					experimental = {
						useFlatConfig = true,
					},
				},
				root_dir = lspconfig.util.root_pattern("eslint.config.js"),
			})
			lspconfig.denols.setup({
				root_dir = lspconfig.util.root_pattern("deno.json", "deno.jsonc"),
			})
			lspconfig.coffeesense.setup({})
			lspconfig.fish_lsp.setup({})
			lspconfig.svelte.setup({
				root_dir = lspconfig.util.root_pattern("svelte.config.js"),
			})
			lspconfig.graphql.setup({})
			lspconfig.ruby_lsp.setup({})
			lspconfig.lua_ls.setup({})
			lspconfig.nim_langserver.setup({})
		end,
	},
```
What am I doing wrong? (And my full [plugins.lua](https://github.com/ruiiiijiiiiang/dotfiles/blob/master/nvim/.config/nvim/lua/plugins/plugins.lua#L143-L180) for context)

**Reply by dpetka2001** [reply]
That's not a Lazyvim installation. You just tried to incorporate Lazyvim into your configuration. Start with a fresh Lazyvim installation and read the docs on the website about how things should be configured. And if you have any further questions regarding a Lazyvim installation feel free to ask.

**Reply by ruiiiijiiiiang** [reply]
> That's not a Lazyvim installation. You just tried to incorporate Lazyvim into your configuration. Start with a fresh Lazyvim installation and read the docs on the website about how things should be configured. And if you have any further questions regarding a Lazyvim installation feel free to ask.

Sorry I only committed files that I've changed to my dotfile repo. But I did install everything with lazyvim.
![nvim config](https://github.com/user-attachments/assets/84498a01-814b-4e83-90f3-56c2f4b4f5a0)

**Reply by folke** [reply, marked as answer]
you are ovcerriding the whole LAzyVim lspconfig config function, so yes that won't work

**Reply by ruiiiijiiiiang** [reply]
> you are ovcerriding the whole LAzyVim lspconfig config function, so yes that won't work

Ohhhh I had no idea I was doing that. After I got rid of this block, everything is working now. Thank you so much!
